File: baekjoon/자료 구조/[10845]큐.py
from collections import deque
import sys

n = int(sys.stdin.readline())

queue = deque()

for i in range(n):
    command = sys.stdin.readline().split()
    
    if command[0] == "push":
        queue.append(command[1])
    if command[0] == "pop":
        if len(queue) == 0:
            print(-1)
        else:
            print(queue.popleft())
    if command[0] == "size":
        print(len(queue))
    if command[0] == "empty":
        if len(queue) == 0:
            print(1)
        else:
            print(0)
    if command[0] == "front":
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[0])
    if command[0] == "back":
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[-1])

File: baekjoon/자료 구조/[10845]큐.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <queue>
using namespace std;

int main() {
    int n;
    queue <int> q;
    string com;

    cin >> n;

    for (int i = 0; i < n; i++) {
        cin >> com;
        if (com == "push") {
            int temp;
            cin >> temp;
            q.push(temp);
        }
        else if (com == "pop") {
            if (!q.empty()) {
                cout << q.front() << "\n";
                q.pop();
            }
            else cout << -1 << "\n";
        }
        else if (com == "size") cout << q.size() << "\n";

        else if (com == "empty") cout << q.empty() << "\n";

        else if (com == "front") {
            if (!q.empty()) cout << q.front() << "\n";
            else cout << -1 << "\n";
        }

        else if (com == "back") {
            if (!q.empty()) cout << q.back() << "\n";
            else cout << -1 << "\n";
        }
    }

    return 0;
}